lib/hooks/useSearchTrackedRepos.ts

```diff
@@ -0,0 +1,47 @@
+import useSWR, { Fetcher } from "swr";
+import useDebounceTerm from "./useDebounceTerm";
+
+const apiFetcher: Fetcher = async (endpointString: string) => {
+  const res = await fetch(`${process.env.NEXT_PUBLIC_API_URL}/${endpointString}`, {
+    headers: { accept: "application/json" },
+  });
+
+  if (res.status === 404) {
+    const error = new Error("HttpError");
+
+    error.message = `${res.status} ${res.statusText}`;
+    error.stack = JSON.stringify(await res.json());
+    // eslint-disable-next-line no-console
+    console.error(error);
+
+    throw error;
+  }
+
+  return res.json();
+};
```

Comment:
Any reason we can't use the public API fetcher utility here?

https://github.com/open-sauced/app/blob/314f51e3ab3f5b1310662f060d1737f734479872/lib/utils/public-api-fetcher.ts#L10-L12

Reply:
no specific reason, I was modeling this based on another that wasn't using the publicApiFetcher, I'll change it! 

To be frank however, I thought the requirements for making this work would need the `searchTrackedRepos`, but it didn't since the page grabs all the repos as `PagedData`. There's probably an edge cases for large lists (like hundreds or thousands) that would need it, but otherwise this can be punted to later.